src/rust/en.omegascans/src/parser.rs

```diff
@@ -0,0 +1,233 @@
+use aidoku::{
+	error::Result, helpers::uri::encode_uri_component, prelude::format, std::net::HttpMethod,
+	std::net::Request, std::String, std::Vec, Chapter, Filter, FilterType, Manga,
+	MangaContentRating, MangaPageResult, MangaStatus, MangaViewer, Page,
+};
+
+const BASE_API_URL: &str = "https://api.omegascans.org";
+
+pub fn parse_manga_list(base_url: String, filters: Vec<Filter>) -> Result<MangaPageResult> {
+	let mut search_query = String::new();
+	let mut status = String::new();
+	let mut genres = String::new();
+
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				if let Ok(filter_value) = filter.value.as_string() {
+					search_query.push_str(&encode_uri_component(filter_value.read()));
+				}
+			}
+			FilterType::Genre => {
+				if let Ok(filter_id) = filter.object.get("id").as_string() {
+					match filter.value.as_int().unwrap_or(-1) {
+						1 => {
+							genres.push_str(filter_id.read().as_str());
+							genres.push(',');
+						}
+						_ => continue,
+					}
+				}
+			}
+			FilterType::Select => match filter.name.as_str() {
+				"Status" => match filter.value.as_int().unwrap_or(-1) {
+					0 => status.push_str("All"),
+					1 => status.push_str("Ongoing"),
+					2 => status.push_str("Hiatus"),
+					3 => status.push_str("Dropped"),
+					4 => status.push_str("Completed"),
+					5 => status.push_str("Cancelled"),
+					_ => continue,
+				},
+				_ => continue,
+			},
+			_ => continue,
+		}
+	}
+
+	if !genres.is_empty() {
+		genres.pop();
+	}
+
+	let url = format!("{}/query?query_string={}&series_status={}&order=desc&orderBy=total_views&series_type=Comic&page=1&perPage=1000&tags_ids=[{}]", BASE_API_URL, search_query, status, genres);
+	let json = Request::new(url, HttpMethod::Get);
+	let mangas = parse_manga(base_url.clone(), json)?;
+
+	Ok(MangaPageResult {
+		manga: mangas,
```

Comment:
you should just call the variable manga and then you can do this (plural of manga is manga)
```suggestion
	let manga = parse_manga(base_url.clone(), json)?;

	Ok(MangaPageResult {
		manga,
```